File: frontend/lib/profileService.ts
import api from "./api";
import type { ApiResponse, ProfilePictureResponse } from "./types";

export const profileService = {
  // Upload user profile picture
  async uploadUserProfilePicture(file: File) {
    const formData = new FormData();
    formData.append("file", file);

    const res = await api.post<ApiResponse<ProfilePictureResponse>>(
      "/profile/picture/upload",
      formData,
      {
        headers: {
          "Content-Type": "multipart/form-data",
        },
      },
    );
    return res.data.data;
  },

  // Upload trainer profile picture
  async uploadTrainerProfilePicture(trainerId: number, file: File) {
    const formData = new FormData();
    formData.append("file", file);

    const res = await api.post<ApiResponse<ProfilePictureResponse>>(
      `/profile/trainer/${trainerId}/picture/upload`,
      formData,
      {
        headers: {
          "Content-Type": "multipart/form-data",
        },
      },
    );
    return res.data.data;
  },
};

File: backend/src/main/java/com/sofen/backend/features/profile/service/ProfileService.java
package com.sofen.backend.features.profile.service;

import com.sofen.backend.features.profile.dto.response.ProfilePictureResponse;
import org.springframework.web.multipart.MultipartFile;

public interface ProfileService {
    ProfilePictureResponse uploadUserProfilePicture(Long userId, MultipartFile file);

    ProfilePictureResponse uploadTrainerProfilePicture(Long trainerId, MultipartFile file);
}
